packages/worker/src/spot-termination.ts

```diff
@@ -0,0 +1,35 @@
+import {
+	changeMessageVisibility,
+	logger,
+} from '@guardian/transcription-service-backend-common';
+import { SQSClient } from '@aws-sdk/client-sqs';
+import { getCurrentReceiptHandle, setInterruptionTime } from './index';
+
+const CHECK_FREQUENCY = 10;
+
+export const checkSpotInterrupt = async (
+	client: SQSClient,
+	queueUrl: string,
+) => {
+	const url = 'http://169.254.169.254/latest/meta-data/spot/instance-action';
+	const result = await fetch(url);
+	if (result.status === 200) {
+		const json = await result.json();
+		if (json.action === 'terminate') {
+			setInterruptionTime(new Date(json.time));
+			logger.warn('Spot instance termination detected');
+			// Interrupt warning occurs 2 minutes before termination
+			const receiptHandle = getCurrentReceiptHandle();
+			if (!receiptHandle) {
+				return;
+			}
+			await changeMessageVisibility(client, queueUrl, receiptHandle, 110);
+			// once the interrupt warning has happened, we don't need to keep checking
+			return;
+		}
```

Comment:
it might be too complicated to change in this PR but at the moment I think when a very unlucky job is picked up by three successive workers which are interrupted before finished transcription, we will put the message on the dead letter queue without notifying the user. 
We could use the two minutes before termination to send a failure message. 
Maybe for the moment, just having the alarm when the message is added to the dead letter queue is enough and we can find out as people use the service how common this case actually is.